tests/models/anthropic/test_output.py

```diff
@@ -0,0 +1,705 @@
+    assert 'strict' not in tools[0]
+    # Schema still has the constraint (not removed)
+    assert tools[0]['input_schema']['properties']['username']['minLength'] == 3
+    assert betas is OMIT
+
+
+# =============================================================================
+# NATIVE OUTPUT - Model Support
+# =============================================================================
+
+
+def test_native_output_supported_model(
+    allow_model_requests: None,
+    mock_sonnet_4_5: tuple[AnthropicModel, AsyncAnthropic],
+    city_location_schema: type[BaseModel],
+):
+    """sonnet-4-5: NativeOutput → strict=True + beta header + output_format."""
+    model, mock_client = mock_sonnet_4_5
+    agent = Agent(model, output_type=NativeOutput(city_location_schema))
+
+    agent.run_sync('What is the capital of France?')
+
+    completion_kwargs = get_mock_chat_completion_kwargs(mock_client)[-1]
+    output_format = completion_kwargs['output_format']
+    betas = completion_kwargs['betas']
+
+    assert output_format['type'] == 'json_schema'
+    assert output_format['schema']['type'] == 'object'
+    assert betas == snapshot(['structured-outputs-2025-11-13'])
+
+
+def test_native_output_unsupported_model_raises_error(
+    allow_model_requests: None, city_location_schema: type[BaseModel]
+):
+    """sonnet-4-0: NativeOutput → raises UserError."""
+    mock_client = MockAnthropic.create_mock(
+        completion_message([BetaTextBlock(text='test', type='text')], BetaUsage(input_tokens=5, output_tokens=2))
+    )
+    model = AnthropicModel('claude-sonnet-4-0', provider=AnthropicProvider(anthropic_client=mock_client))
+    agent = Agent(model, output_type=NativeOutput(city_location_schema))
+
+    with pytest.raises(UserError, match='Model claude-sonnet-4-0 does not support native output.'):
+        agent.run_sync('What is the capital of France?')
+
+
+# =============================================================================
+# AUTO MODE Selection
+# =============================================================================
+
+
+def test_auto_mode_model_profile_check(allow_model_requests: None):
+    """Verify profile.supports_json_schema_output is set correctly."""
+    mock_client = MockAnthropic.create_mock(
+        completion_message([BetaTextBlock(text='test', type='text')], BetaUsage(input_tokens=5, output_tokens=2))
+    )
+
+    sonnet_4_5 = AnthropicModel('claude-sonnet-4-5', provider=AnthropicProvider(anthropic_client=mock_client))
+    assert sonnet_4_5.profile.supports_json_schema_output is True
+
+    sonnet_4_0 = AnthropicModel('claude-sonnet-4-0', provider=AnthropicProvider(anthropic_client=mock_client))
+    assert sonnet_4_0.profile.supports_json_schema_output is False
+
+
+# =============================================================================
+# BETA HEADER Management
+# =============================================================================
+
+
+def test_beta_header_merge_custom_headers(
```

Comment:
This test is not specifically about output, so it may be better on the main anthropic tests. Especially if we can combine it with a builtin tool that needs a beta header, and ensure we end up with all of them